lambda/models/state_machine/create_model.py

```diff
@@ -16,25 +16,62 @@
 
 from copy import deepcopy
 from typing import Any, Dict
+import json
+import os
+
+import boto3
+from botocore.exceptions import ClientError
 
 
 def handle_set_model_to_creating(event: Dict[str, Any], context: Any) -> Dict[str, Any]:
     """Set DDB entry to CREATING status."""
     output_dict = deepcopy(event)
-    output_dict["create_infra"] = True
+    output_dict["create_infra"] = 'modelConfig' in event
     return output_dict
 
 
 def handle_start_copy_docker_image(event: Dict[str, Any], context: Any) -> Dict[str, Any]:
     """Start process for copying Docker image into local AWS account."""
     output_dict = deepcopy(event)
+
+    lambdaClient = boto3.client("lambda")
+
+    response = lambdaClient.invoke(
+        FunctionName=os.environ["DOCKER_IMAGE_BUILDER_FN_ARN"],
+        Payload=json.dumps(event['imageConfig'])
+    )
+
+    payload = response["Payload"].read()
+    print(f"Got payload {payload}")
+    output_dict["image_info"] = json.loads(payload)
     return output_dict
 
 
 def handle_poll_docker_image_available(event: Dict[str, Any], context: Any) -> Dict[str, Any]:
     """Check that Docker image is available in account or not."""
     output_dict = deepcopy(event)
+
+    ecrClient = boto3.client("ecr")
+    try:
+        response = ecrClient.describe_images(
+            repositoryName=os.environ["ECR_REPOSITORY_NAME"],
+            imageIds=[
+                {
+                    "imageTag": event["image_info"]["image_tag"]
+                }
+            ]
+        )
+    except ClientError as e:
```

Comment:
do we get anything more specific than  a "ClientError" if the image doesn't exist yet? I'd prefer to have the exact exception so we don't unintentionally catch errors that we shouldn't attempt to retry on

Reply:
narrowed down scope to ImageDoesNotExist